isisd/isis_circuit.c

```diff
@@ -1397,6 +1397,14 @@ void isis_circuit_circ_type_set(struct isis_circuit *circuit, int circ_type)
 	if (circuit->circ_type == circ_type)
 		return;
 
+	/* If the interface is passive and we switch from bcast to p2p,
```

Comment:
Please add the detailed breakdown of what is going on here to the commit comment.